src/ldp/AuthenticatedLdpHandler.ts

```diff
@@ -0,0 +1,145 @@
+   */
+  authorizer: Authorizer;
+  /**
+   * Executed the operation.
+   */
+  operationHandler: OperationHandler;
+  /**
+   * Writes out the response of the operation.
+   */
+  responseWriter: ResponseWriter;
+}
+
+/**
+ * The central manager that connects all the necessary handlers to go from an incoming request to an executed operation.
+ */
+export class AuthenticatedLdpHandler implements HttpHandler {
+  private readonly requestParser: RequestParser;
+
+  private readonly credentialsExtractor: CredentialsExtractor;
+
+  private readonly permissionsExtractor: PermissionsExtractor;
+
+  private readonly authorizer: Authorizer;
+
+  private readonly operationHandler: OperationHandler;
+
+  private readonly responseWriter: ResponseWriter;
+
+  /**
+   * Creates the handler.
+   * @param args - The handlers required. None of them are optional.
+   */
+  public constructor (args: AuthenticatedLdpHandlerArgs) {
+    Object.assign(this, args);
+  }
+
+  /**
+   * Checks if the incoming request can be handled. The check is very non-restrictive and will usually be true.
+   * It is based on whether the incoming request can be parsed to an operation.
+   * @param input - Incoming request and response. Only the request will be used.
+   *
+   * @returns A promise resolving if this request can be handled, otherwise rejecting with an Error.
+   */
+  public async canHandle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    return this.requestParser.canHandle(input.request);
+  }
+
+  /**
+   * Handles the incoming request and writes out the response.
+   * This includes the following steps:
+   *  - Parsing the request to an Operation.
+   *  - Extracting credentials from the request.
+   *  - Extracting the required permissions.
+   *  - Validating if this operation is allowed.
+   *  - Executing the operation.
+   *  - Writing out the response.
+   * @param input - The incoming request and response object to write to.
+   *
+   * @returns A promise resolving when the handling is finished.
+   */
+  public async handle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    let err: Error = null;
+    let operation: Operation = null;
+
+    try {
+      operation = await this.runHandlers(input.request);
+    } catch (error) {
+      err = error;
+    }
+
+    const writeData = { response: input.response, operation, error: err };
+
+    await this.responseWriter.canHandle(writeData);
+
+    return this.responseWriter.handle(writeData);
+  }
+
+  /**
+   * Runs all handlers except writing the output to the response.
+   * This because any errors thrown here have an impact on the response.
+   * @param request - Incoming request.
+   *
+   * @returns A promise resolving to the generated Operation.
+   */
+  private async runHandlers (request: HttpRequest): Promise<Operation> {
+    const op: Operation = await this.runHandler(this.requestParser, request);
+
```

Comment:
Are these empty lines also enforced by the linter?
If yes, I would adjust the linter rules (if not done yet in the other PR).
If no, I would still remove them :-)

Reply:
Funnily enough some of them are enforced :D (the ones between variable assignment lines and non-assignment lines). So that would be another one for the eslint pull request.